### ISSUE TITLE
Exception while testing on custom dataset

### ISSUE THREAD

**Issue by maltyyev** [zhangxiangxiao/Crepe#23]
Hi,
I've created my own training and testing datasets with just 2 classes. I changed the number of output classes in config.lua and set it to 2 as per instructions in the related article. Now, the training goes on well, but testing crashes with this exception:

> Testing on training data for era 1
> qlua: /home/murad/torch/install/share/lua/5.1/nn/THNN.lua:110: Assertion `cur_target >= 0 && cur_target < n_classes' failed.  at /home/murad/torch/extra/nn/lib/THNN/generic/ClassNLLCriterion.c:57
> stack traceback:
> 	[C]: at 0x7f0b7de7ef50
> 	[C]: in function 'v'
> 	/home/murad/torch/install/share/lua/5.1/nn/THNN.lua:110: in function 'ClassNLLCriterion_updateOutput'
> 	...rad/torch/install/share/lua/5.1/nn/ClassNLLCriterion.lua:44: in function 'forward'
> 	./test.lua:60: in function 'run'
> 	main.lua:145: in function 'run'
> 	main.lua:42: in function 'main'
> 	main.lua:316: in main chunk

I googled it out and read that it means that the number of output classes in the last layer doesn't match the number of classes in dataset which obviously cannot be, I checked it out stupid amount of times!
I'm running the code without CUDA, in CPU only mode. Will appreciate any help. Thank you.

**Comment by zhangxiangxiao** [comment]
This seems like a weird error. You probably want to check on the dataset file and see if anything strange is going on. Otherwise, write a unittest and try to print what label was given to the model in test.lua.

I will leave this issue open in case anybody else has a good debugging advice.

**Comment by zhangxiangxiao** [comment]
Closing due to inactivity.
